rdt-open-reader/src/main/java/io/ona/rdt/fragment/RDTJsonFormFragment.java

```diff
@@ -231,4 +257,23 @@ public RDTJsonFormActivity getRdtActivity() {
         return (RDTJsonFormActivity) getActivity();
     }
 
+    private String getUniqueId(JSONObject jsonForm) {
+        String uniqueId = null;
+        try {
+            JSONArray fields = jsonForm.getJSONObject("step6").getJSONArray(JsonFormConstants.FIELDS);
+            for (int i = 0; i < fields.length(); i++) {
+                JSONObject field = fields.getJSONObject(i);
+                if ("unique_id".equals(field.getString(JsonFormConstants.KEY))) {
+                    final String value = field.getString(JsonFormConstants.VALUE);
+                    if (StringUtils.isNotBlank(value)) {
+                        uniqueId = value;
+                    }
+                }
+            }
+        } catch (JSONException ex) {
+            Timber.e(ex);
+        }
+
+        return uniqueId;
+    }
```

Comment:
Is this supposed to write the barcode sample id to the confirmation screen? If so, this can be achieved by using the `jsonApi` https://github.com/onaio/rdt-standard/blob/b276a520f238cf907e90742a7d4dfe9fa0d54b71/rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/widget/CovidRDTBarcodeFactory.java#L69.

Get the step to write into from the step state config.